packages/medusa/src/loaders/index.ts

```diff
@@ -167,6 +164,47 @@ export default async ({
     [ContainerRegistrationKeys.MANAGER]: asValue(dataSource.manager),
   })
 
+  container.register("remoteQuery", asValue(null)) // ensure remoteQuery is always registered
+  container.register("medusaApp", asValue(null))
+
+  // Only load non legacy modules, the legacy modules (non migrated yet) are retrieved by the registerModule above
+  if (
+    featureFlagRouter.isFeatureEnabled(IsolateProductDomainFeatureFlag.key) ||
+    featureFlagRouter.isFeatureEnabled(IsolatePricingDomainFeatureFlag.key)
+  ) {
+    mergeModulesConfig(configModule.modules ?? {}, modulesConfig)
+
+    const injectedDependencies = {
+      [ContainerRegistrationKeys.PG_CONNECTION]: container.resolve(
+        ContainerRegistrationKeys.PG_CONNECTION
+      ),
+    }
+
+    const medusaApp = await MedusaApp({
+      modulesConfig,
+      servicesConfig: joinerConfig,
+      remoteFetchData: remoteQueryFetchData(container),
+      injectedDependencies,
+    })
+
+    container.register("medusaApp", asValue(medusaApp))
```

Comment:
comment: this will change with phil pr, only the link will be registered in the container instead of everything returned. That way, we don't end up with registrations that we don't really want (we have a lot at the moment) and if medusa app return more things it will not end up in the container un controlled